lib/rpmsg/rpmsg.c

```diff
@@ -24,12 +24,12 @@
  *
  * @return A unique address
  */
-static uint32_t rpmsg_get_address(unsigned long *bitmap, int size)
+static uint32_t rpmsg_get_address(unsigned long *bitmap, unsigned int start, int size)
 {
 	unsigned int addr = RPMSG_ADDR_ANY;
 	unsigned int nextbit;
 
-	nextbit = metal_bitmap_next_clear_bit(bitmap, 0, size);
+	nextbit = metal_bitmap_loopnext_clear_bit(bitmap, start % size, size);
```

Comment:
regarding libmetal PR
```c
static inline unsigned int
metal_bitmap_loopnext_clear_bit(unsigned long *bitmap, unsigned int start,
				unsigned int max)
{
	unsigned int bit;

	for (bit = start;
	     bit < max + start && !metal_bitmap_is_bit_clear(bitmap, bit % max);
	     bit++)
		;
	return bit % max;
}
```
Not clear to me why you have to use a new metal function
`metal_bitmap_next_clear_bit(bitmap, start % size, size) `does not work?

Reply:
metal_bitmap_next_clear_bit:
find bit from: start ... max

metal_bitmap_loopnext_clear_bit:
find bit from:
start ... max
**0 ... start**


metal_bitmap_loopnext_clear_bit can search back from the 0, when meet the max.

Reply:
May be we can improve current function as it is?

Such as if start is not 0, then loopback after max.